frontend/packages/console-shared/src/hooks/select-list.ts

```diff
@@ -6,12 +6,21 @@ export const useSelectList = <R extends K8sResourceCommon>(
   data: R[],
   visibleRows: Set<string>,
   onRowSelected: (rows: R[]) => void,
+  preSelected: string[] = [],
 ): {
   onSelect: OnSelect;
   selectedRows: Set<string>;
   updateSelectedRows: (rows: R[]) => void;
 } => {
   const [selectedRows, setSelectedRows] = React.useState<Set<string>>(new Set());
+  const prevProps = React.useRef<string[]>(null);
+
+  React.useEffect(() => {
+    if (preSelected.length > 0 && prevProps.current !== preSelected) {
```

Comment:
needs optional chaining since we wont have default value
also this is supposed to be only initial selection we should make sure it runs only once, right ?
```suggestion
    if (preSelected?.length > 0 && prevProps.current !== preSelected && selectedRows.length === 0) {
```

Do you need to store the `prevProps` and check it against `preSelected` ?

Reply:
yes that's the reason we are checking with prevProps